pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,259 @@
+	}
+
+	// Fetch deploymentConfig
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
+	}
+	// See latest version; if 0, there are no deployments for this config
+	desiredVersion := config.LatestVersion
+	if desiredVersion == 0 {
+		return nil, errors.NewBadRequest(fmt.Sprintf("no deployment exists for deploymentConfig %q", config.Name))
+	}
+
+	// Support retrieving logs for older deployments
+	switch {
+	case deployLogOpts.Version == -1:
+		// Latest
+	case deployLogOpts.Version <= 0 || deployLogOpts.Version > config.LatestVersion:
+		// Invalid version
+		return nil, errors.NewBadRequest(fmt.Sprintf("invalid version for deploymentConfig %q: %d", config.Name, deployLogOpts.Version))
+	default:
+		desiredVersion = deployLogOpts.Version
+	}
+
+	// List deployments
+	deployments, err := r.Deploy.ListDeployments(ctx, deployutil.ConfigSelector(config.Name))
+	if err != nil {
+		return nil, err
+	}
+	if len(deployments.Items) == 0 {
+		return nil, errors.NewInternalError(fmt.Errorf("cannot find deployments for deploymentConfig %q", config.Name))
+	}
+	sort.Sort(deployutil.DeploymentsByLatestVersionDesc(deployments.Items))
+
+	// Find desired deployment
+	var target *kapi.ReplicationController
+	for _, deployment := range deployments.Items {
+		if desiredVersion == deployutil.DeploymentVersionFor(&deployment) {
+			target = &deployment
+			break
+		}
+	}
+	if target == nil {
+		desiredDeployment := deployutil.VersionedLabelForDeploymentConfig(config, desiredVersion)
+		glog.V(3).Infof("Cannot find desired deployment (%s) for %s", desiredDeployment, config.Name)
+		return nil, errors.NewNotFound("deployment", desiredDeployment)
+	}
+
+	// Check for deployment status; if it is new or pending, we will wait for it. If it is complete,
+	// the deployment completed successfully and the deployer pod will be deleted so we will return a
+	// success message. If it is running or failed, retrieve the log from the deployer pod.
+	status := deployutil.DeploymentStatusFor(target)
+	switch status {
+	case deployapi.DeploymentStatusNew, deployapi.DeploymentStatusPending:
+		if deployLogOpts.NoWait {
+			glog.V(4).Infof("Deployment %s is in %s state. No logs to retrieve yet.", deployutil.LabelForDeployment(target), status)
+			return &genericrest.LocationStreamer{}, nil
+		}
+		glog.V(4).Infof("Deployment %s is in %s state, waiting for it to start...", deployutil.LabelForDeployment(target), status)
+		lw := createDeploymentLW(ctx, r.Deploy, target)
+		if err := WaitForResource(&kapi.ReplicationController{}, lw, handleDeployment); err != nil {
+			return nil, err
+		}
+	case deployapi.DeploymentStatusComplete:
+		// Deployer pod has been deleted, no logs to retrieve
+		glog.V(4).Infof("Deployment %s was successful so the deployer pod is deleted. No logs to retrieve.", deployutil.LabelForDeployment(target))
+		return &genericrest.LocationStreamer{}, nil
```

Comment:
We know that since the deployment is complete, the deployer pod is deleted so this isn't really a "NotFound" case.